vcpkg/maintainers/registries.md

```diff
@@ -1,281 +1,289 @@
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| `baseline` | string | A string corresponding to the latest available version of the port in the registry. |
+| `port-version` | integer | An integer corresponding to the latest port version of the port in the registry |
 
-Here's an example of a valid baseline.json:
+#### Example of a `baseline.json` file in a Git registry
+
+In a registry containing a single port named `foo` at version `1.0.0#1`, the
+`baseline.json` file contents should be:
 
 ```json
 {
   "default": {
-    "kitten": {
-      "baseline": "2.6.2",
-      "port-version": 0
-    },
-    "port-b": {
-      "baseline": "19.00",
-      "port-version": 2
+    "foo": { 
+      "baseline": "1.0.0", 
+      "port-version": "1"
     }
   }
 }
 ```
 
-The `versions` directory contains all the information about which versions of which packages are contained in the registry, along with where those versions are stored. The rest of the registry just acts as a backing store, as far as vcpkg is concerned: only things inside the `versions` directory will be used to direct how your registry is seen by vcpkg.
+### <a name="git-version-file"></a> Version file layout in Git registries
+
+The `versions` directory contains all the information about which versions of
+packages are contained in the registry, along with the method to retrieve those
+versions from the repository's history.
+
+*Top-level fields*
 
-Each port in a registry should exist in the versions directory as `<first letter of port>-/<name of port>.json`; in other words, the information about the `kitten` port would be located in `versions/k-/kitten.json`. This should be a top-level object with only a single field: `"versions"`. This field should contain an array of version objects:
+| Name       | Type            | Description |
+| ---------- | --------------- | --------------------------- |
+| `versions` | VersionObject[] | An array of version objects. Contains an entry for each version of the port in the history of the registry. |
 
-- The version of the port in question; should be exactly the same as the `vcpkg.json` file, including the version fields and `"port-version"`.
-- The `"git-tree"` field, which is a git tree; in other words, what you get when you write `git rev-parse COMMIT-ID:path/to/port`.
+*VersionObject*
 
-The version field for ports with deprecated `CONTROL` files is `"version-string"`.
+| Name       | Type   | Description |
+| ---------- | ------ | ----------- |
+| `git-tree` | string | A Git calculated SHA used to retrieve the port version |
+| [`version`<br>`version-semver`<br>`version-date`<br>`version-string`](../reference/vcpkg-json.md#version) | string | Upstream version information |
+| [port-version](../reference/vcpkg-json.md#port-version) | integer | Port files revision |
 
```

Comment:
I think we should add something like:

> The `version` / `verison-semver` / `version-date` / `version-string` and `port-version` match the values in the port retrieved by the `git-tree`. For example:
> ```console
> PS D:\vcpkg> Get-Content D:\vcpkg\versions\c-\curl.json -TotalCount 7
> {
>   "versions": [
>     {
>       "git-tree": "6ef1763f3cbe570d6378632c9b5793479c37fb07",
>       "version": "8.11.0",
>       "port-version": 1
>     },
> PS D:\vcpkg> git show 6ef1763f3cbe570d6378632c9b5793479c37fb07:vcpkg.json | Select-Object -First 5
> {
>   "name": "curl",
>   "version": "8.11.0",
>   "port-version": 1,
>   "description": "A library for transferring data with URLs",
> ```

I'm torn on whether this is a 'must' situation. It's true that if a user is manually doing this they must ensure that they do this, but it's also true that we provide tooling that makes this an implementation detail most of the time.

Reply:
Maybe we can add this to the "Recommended practices" section in the conceptual article.